components/exercises/chapter-5-oscar/Flattening/Flattening.tsx

```diff
@@ -0,0 +1,66 @@
+import { getFlattenedArray } from './getFlattenedArray';
+
+const arrays = [
+  ['❤️✨', '🚀🌟', '💭💪'],
+  ['🌅🔄', '💪😃'],
+  ['🚫🙅‍♂️', '💪🏆'],
+];
+
+const phrases = [
+  { icon: '❤️✨', phrase: 'Believe in yourself, and you are halfway there.' },
+  { icon: '🚀🌟', phrase: 'Success is a journey, not a destination.' },
+  { icon: '💭💪', phrase: 'Dream big and dare to fail.' },
+  { icon: '🌅🔄', phrase: 'Every setback is a setup for a comeback.' },
+  { icon: '💪😃', phrase: 'You are stronger than you think.' },
+  { icon: '🚫🙅‍♂️', phrase: 'The only limit is the one you set for yourself.' },
+  {
+    icon: '💪🏆',
+    phrase: 'Hard work beats talent when talent does not work hard.',
+  },
+];
+
+export default function Flattening(): JSX.Element {
+  /**
+   * Retrieves the flattened array of phrases from the given array of arrays, and maps each phrase to a JSX element containing the corresponding icon and phrase.
+   * @param arrays - The array of arrays to be flattened.
+   * @returns An array of JSX elements representing the flattened phrases.
+   */
+  const getPhrase = getFlattenedArray(arrays).map((icon, index) => {
+    const matchingIcon = phrases.find((phrase) => phrase.icon === icon);
+    return (
+      matchingIcon && (
+        <div
+          className='m-2 rounded-sm bg-white p-4'
+          data-testid={`phrase-${matchingIcon.icon}`}
+          key={matchingIcon.icon}
+        >
+          {matchingIcon.phrase}
+        </div>
+      )
+    );
+  });
+
+  return (
+    <div className='flex flex-col items-center'>
+      <div className='max-w-lg rounded-md bg-blue-300'>
+        <div className='flex p-4'>
+          <div className=' flex justify-between gap-2 rounded-sm'>
+            {arrays.map((icons) => {
+              return (
+                <span
+                  className='rounded-md bg-white p-3'
+                  key={icons.toString()}
+                >
+                  <span>{icons}</span>
+                </span>
+              );
+            })}
+          </div>
+        </div>
+      </div>
+      <div className='mt-2 max-w-lg rounded-md bg-green-300'>
+        <div className='flex flex-col gap-2 p-4'>{getPhrase}</div>
+      </div>
+    </div>
+  );
+}
```

Comment:
I like the idea and at the same time I didn't understand how it worked until I saw the code, so I'm afraid this could happen to anyone visiting the website...

Here's an idea to make more obvious how this works: what about rendering each couple of icons next to the phrase they correspond to? So it is obvious that we flattened the array and then used a couple of icons to render each phrase.

Reply:
thank you for the suggestion I have added the icons